### PR TITLE
fix: Correct AtlasMapping datasource URI for XSD document types and c…

### PR DESCRIPTION
**Pull request by pleacu**
…orrect JAVA detection

Fixes: #817 
This corrects the error:
2019-03-05 10:14:26.035 ERROR 28221 --- [  XNIO-2 task-1] io.atlasmap.core.DefaultAtlasContext     : Cannot find module info for the DataSource uri 'atlas:xsd:fhir-rafal-170'

It's odd that the DocumentType.JAVA enum was 'Java' instead of 'JAVA'.  Also added some defensive code in the RT service (pulled a NPE at one point).

### CODE REVIEW

File: ui/src/app/lib/atlasmap-data-mapper/models/config.model.ts [hunk 1]
```diff
@@ -197,9 +197,12 @@ export class ConfigModel {
     docDef.inspectionSource = docInitModel.inspectionSource;
     docDef.inspectionResult = docInitModel.inspectionResult;
     docDef.selectedRoot = docInitModel.selectedRoot;
-    docDef.uri = 'atlas:' + docDef.type.toLowerCase() + ':' + docDef.id;
-    if (docDef.type === DocumentType.JAVA) {
+    if (docDef.type === DocumentType.XSD) {
+      docDef.uri = 'atlas:' + 'xml' + ':' + docDef.id;
+    } else if (docDef.type === DocumentType.JAVA) {
       docDef.uri += '?className=' + docDef.inspectionSource;
```
Comment (igarashitm) [inline ui/src/app/lib/atlasmap-data-mapper/models/config.model.ts]: isn't it going to miss `atlas:java:id` prefix?

File: ui/src/app/lib/atlasmap-data-mapper/common/config.types.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@
 */
 
 export enum DocumentType {
-  JAVA = 'Java',
+  JAVA = 'JAVA',
   JAVA_ARCHIVE = 'JAR',
```
Comment (igarashitm) [inline ui/src/app/lib/atlasmap-data-mapper/common/config.types.ts]: is it possible JAVA_ARCHIVE to be used for uri, causes `atlas:jar'?

Comment (pleacu) [reply ui/src/app/lib/atlasmap-data-mapper/common/config.types.ts]: Is  atlas:jar:id valid?

Comment (igarashitm) [reply ui/src/app/lib/atlasmap-data-mapper/common/config.types.ts]: nope, it should be `atlas:java:id` as it's still supposed to be handled with JavaModule.

Comment (pleacu) [reply ui/src/app/lib/atlasmap-data-mapper/common/config.types.ts]: I'll push the fix for the first comment - if atlas.jar is required we can catch that in 7.4

Comment (igarashitm) [reply ui/src/app/lib/atlasmap-data-mapper/common/config.types.ts]: ah no we don't need it, I just worry if in any case DocumentType.JAVA_ARCHIVE is passed into ConfigModel, and then it causes "atlas:jar:id" uri which is wrong.

File: ui/src/app/lib/atlasmap-data-mapper/models/config.model.ts (hunk 1 above)
Comment (igarashitm) [reply ui/src/app/lib/atlasmap-data-mapper/models/config.model.ts]: So actually wondering if we should add `atlas:java:id` when `docDef.type === DocumentType.JAVA` **OR** `docDef.type === DocumentType.JAVA_ARCHIVE`. I'm not sure if it's possible at all to get `DocumentType.JAVA_ARCHIVE` passed in here.

Comment (pleacu) [reply ui/src/app/lib/atlasmap-data-mapper/models/config.model.ts]: so - this is what we've got now:
    if (docDef.type === DocumentType.XSD) {
      docDef.uri = 'atlas:' + 'xml' + ':' + docDef.id;
    } else {
      docDef.uri = 'atlas:' + docDef.type.toLowerCase() + ':' + docDef.id;
    }

    if (docDef.type === DocumentType.JAVA) {
      docDef.uri += '?className=' + docDef.inspectionSource;
    }

So - is this what you are suggesting?
    if (docDef.type === DocumentType.XSD) {
      docDef.uri = 'atlas:' + 'xml' + ':' + docDef.id;
    } else {
      docDef.uri = 'atlas:' + docDef.type.toLowerCase() + ':' + docDef.id;
    }

    if (docDef.type === DocumentType.JAVA ||docDef.type === DocumentType.JAVA_ARCHIVE ) {
      docDef.uri += '?className=' + docDef.inspectionSource;
    }

Comment (igarashitm) [reply ui/src/app/lib/atlasmap-data-mapper/models/config.model.ts]: I would suggest this

```java
if (docDef.type === DocumentType.XSD) {
  docDef.uri = 'atlas:' + 'xml' + ':' + docDef.id;
} else if (docDef.type === DocumentType.JAVA_ARCHIVE) {
  docDef.uri = 'atlas:' + 'java' + ':' + docDef.id;
} else {
  docDef.uri = 'atlas:' + docDef.type.toLowerCase() + ':' + docDef.id;
}

if (docDef.type === DocumentType.JAVA ||docDef.type === DocumentType.JAVA_ARCHIVE ) {
  docDef.uri += '?className=' + docDef.inspectionSource;
}
```

might be even better splitting the enum into two, module type (java, json, xml) and others (schema, instance, etc) but maybe next time

**Pull request merged**